Bottom: faa1317e1251b40eab14c2d4bd0b8c739c26e6da
Top:    b3626f9279513682eaea584997a80902d06e513f
Author: Ian Kent <raven@themaw.net>
Date:   2013-06-07 18:13:45 +0800

Refresh of debug.patch

---

diff --git a/lib/cache.c b/lib/cache.c
index 23462a2..ecace4a 100644
--- a/lib/cache.c
+++ b/lib/cache.c
@@ -658,10 +658,8 @@ int cache_update_offset(struct mapent_cache *mc, const char *mkey, const char *k
 		return CHE_FAIL;
 
 	me = cache_lookup_distinct(mc, key);
-	if (me)
-		error(LOGOPT_ANY, "owner %p owner->key %s me %p me->key %s", owner,owner->key,me,me->key);
 	if (me && me->age == age) {
-		if (me == owner)
+		if (me == owner || strcmp(me->key, key) == 0)
 			return CHE_DUPLICATE;
 	}
